TST: 3 tests failed because the key_press_event function was deprecated in Matplotlib 3.6

Example log: https://app.circleci.com/pipelines/github/astropy/astropy/11465/workflows/e2afffbb-6ad6-4b02-ae10-5d2e8e891b1c/jobs/102136

```
Matplotlib: 3.6.0.dev2720+g330124862
...
________________ TestDisplayWorldCoordinate.test_overlay_coords ________________

self = <astropy.visualization.wcsaxes.tests.test_display_world_coordinates.TestDisplayWorldCoordinate object at 0x7f44b0db1580>
ignore_matplotlibrc = None
tmpdir = local('/tmp/pytest-of-circleci/pytest-0/test_overlay_coords0')

    def test_overlay_coords(self, ignore_matplotlibrc, tmpdir):
        wcs = WCS(self.msx_header)
    
        fig = plt.figure(figsize=(4, 4))
        canvas = fig.canvas
    
        ax = WCSAxes(fig, [0.1, 0.1, 0.8, 0.8], wcs=wcs)
        fig.add_axes(ax)
    
        # On some systems, fig.canvas.draw is not enough to force a draw, so we
        # save to a temporary file.
        fig.savefig(tmpdir.join('test1.png').strpath)
    
        # Testing default displayed world coordinates
        string_world = ax._display_world_coords(0.523412, 0.518311)
        assert string_world == '0\xb029\'45" -0\xb029\'20" (world)'
    
        # Test pixel coordinates
        event1 = KeyEvent('test_pixel_coords', canvas, 'w')
>       fig.canvas.key_press_event(event1.key, guiEvent=event1)

../../.tox/py39-test-image-mpldev/lib/python3.9/site-packages/astropy/visualization/wcsaxes/tests/test_display_world_coordinates.py:44: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
../../.tox/py39-test-image-mpldev/lib/python3.9/site-packages/matplotlib/_api/deprecation.py:199: in wrapper
    emit_warning()
../../.tox/py39-test-image-mpldev/lib/python3.9/site-packages/matplotlib/_api/deprecation.py:193: in emit_warning
    warn_deprecated(
../../.tox/py39-test-image-mpldev/lib/python3.9/site-packages/matplotlib/_api/deprecation.py:96: in warn_deprecated
    warn_external(warning, category=MatplotlibDeprecationWarning)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

message = MatplotlibDeprecationWarning("The key_press_event function was deprecated in Matplotlib 3.6 and will be removed two minor releases later. Use callbacks.process('key_press_event', KeyEvent(...)) instead.")
category = <class 'matplotlib._api.deprecation.MatplotlibDeprecationWarning'>

    def warn_external(message, category=None):
        """
        `warnings.warn` wrapper that sets *stacklevel* to "outside Matplotlib".
    
        The original emitter of the warning can be obtained by patching this
        function back to `warnings.warn`, i.e. ``_api.warn_external =
        warnings.warn`` (or ``functools.partial(warnings.warn, stacklevel=2)``,
        etc.).
        """
        frame = sys._getframe()
        for stacklevel in itertools.count(1):  # lgtm[py/unused-loop-variable]
            if frame is None:
                # when called in embedded context may hit frame is None
                break
            if not re.match(r"\A(matplotlib|mpl_toolkits)(\Z|\.(?!tests\.))",
                            # Work around sphinx-gallery not setting __name__.
                            frame.f_globals.get("__name__", "")):
                break
            frame = frame.f_back
>       warnings.warn(message, category, stacklevel)
E       matplotlib._api.deprecation.MatplotlibDeprecationWarning: The key_press_event function was deprecated in Matplotlib 3.6 and will be removed two minor releases later. Use callbacks.process('key_press_event', KeyEvent(...)) instead.

../../.tox/py39-test-image-mpldev/lib/python3.9/site-packages/matplotlib/_api/__init__.py:363: MatplotlibDeprecationWarning
_________________ TestDisplayWorldCoordinate.test_cube_coords __________________

self = <astropy.visualization.wcsaxes.tests.test_display_world_coordinates.TestDisplayWorldCoordinate object at 0x7f44b0db13a0>
ignore_matplotlibrc = None
tmpdir = local('/tmp/pytest-of-circleci/pytest-0/test_cube_coords0')

    def test_cube_coords(self, ignore_matplotlibrc, tmpdir):
        wcs = WCS(self.cube_header)
    
        fig = plt.figure(figsize=(4, 4))
        canvas = fig.canvas
    
        ax = WCSAxes(fig, [0.1, 0.1, 0.8, 0.8], wcs=wcs, slices=('y', 50, 'x'))
        fig.add_axes(ax)
    
        # On some systems, fig.canvas.draw is not enough to force a draw, so we
        # save to a temporary file.
        fig.savefig(tmpdir.join('test.png').strpath)
    
        # Testing default displayed world coordinates
        string_world = ax._display_world_coords(0.523412, 0.518311)
        assert string_world == '3h26m52.0s 30\xb037\'17\" 2563 (world)'
    
        # Test pixel coordinates
        event1 = KeyEvent('test_pixel_coords', canvas, 'w')
>       fig.canvas.key_press_event(event1.key, guiEvent=event1)

../../.tox/py39-test-image-mpldev/lib/python3.9/site-packages/astropy/visualization/wcsaxes/tests/test_display_world_coordinates.py:124: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
../../.tox/py39-test-image-mpldev/lib/python3.9/site-packages/matplotlib/_api/deprecation.py:199: in wrapper
    emit_warning()
../../.tox/py39-test-image-mpldev/lib/python3.9/site-packages/matplotlib/_api/deprecation.py:193: in emit_warning
    warn_deprecated(
../../.tox/py39-test-image-mpldev/lib/python3.9/site-packages/matplotlib/_api/deprecation.py:96: in warn_deprecated
    warn_external(warning, category=MatplotlibDeprecationWarning)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

message = MatplotlibDeprecationWarning("The key_press_event function was deprecated in Matplotlib 3.6 and will be removed two minor releases later. Use callbacks.process('key_press_event', KeyEvent(...)) instead.")
category = <class 'matplotlib._api.deprecation.MatplotlibDeprecationWarning'>

    def warn_external(message, category=None):
        """
        `warnings.warn` wrapper that sets *stacklevel* to "outside Matplotlib".
    
        The original emitter of the warning can be obtained by patching this
        function back to `warnings.warn`, i.e. ``_api.warn_external =
        warnings.warn`` (or ``functools.partial(warnings.warn, stacklevel=2)``,
        etc.).
        """
        frame = sys._getframe()
        for stacklevel in itertools.count(1):  # lgtm[py/unused-loop-variable]
            if frame is None:
                # when called in embedded context may hit frame is None
                break
            if not re.match(r"\A(matplotlib|mpl_toolkits)(\Z|\.(?!tests\.))",
                            # Work around sphinx-gallery not setting __name__.
                            frame.f_globals.get("__name__", "")):
                break
            frame = frame.f_back
>       warnings.warn(message, category, stacklevel)
E       matplotlib._api.deprecation.MatplotlibDeprecationWarning: The key_press_event function was deprecated in Matplotlib 3.6 and will be removed two minor releases later. Use callbacks.process('key_press_event', KeyEvent(...)) instead.

../../.tox/py39-test-image-mpldev/lib/python3.9/site-packages/matplotlib/_api/__init__.py:363: MatplotlibDeprecationWarning
__________ TestDisplayWorldCoordinate.test_cube_coords_uncorr_slicing __________

self = <astropy.visualization.wcsaxes.tests.test_display_world_coordinates.TestDisplayWorldCoordinate object at 0x7f44b0db1190>
ignore_matplotlibrc = None
tmpdir = local('/tmp/pytest-of-circleci/pytest-0/test_cube_coords_uncorr_slicin0')

    def test_cube_coords_uncorr_slicing(self, ignore_matplotlibrc, tmpdir):
    
        # Regression test for a bug that occurred with coordinate formatting if
        # some dimensions were uncorrelated and sliced out.
    
        wcs = WCS(self.cube_header)
    
        fig = plt.figure(figsize=(4, 4))
        canvas = fig.canvas
    
        ax = WCSAxes(fig, [0.1, 0.1, 0.8, 0.8], wcs=wcs, slices=('x', 'y', 2))
        fig.add_axes(ax)
    
        # On some systems, fig.canvas.draw is not enough to force a draw, so we
        # save to a temporary file.
        fig.savefig(tmpdir.join('test.png').strpath)
    
        # Testing default displayed world coordinates
        string_world = ax._display_world_coords(0.523412, 0.518311)
        assert string_world == '3h26m56.6s 30\xb018\'19\" (world)'
    
        # Test pixel coordinates
        event1 = KeyEvent('test_pixel_coords', canvas, 'w')
>       fig.canvas.key_press_event(event1.key, guiEvent=event1)

../../.tox/py39-test-image-mpldev/lib/python3.9/site-packages/astropy/visualization/wcsaxes/tests/test_display_world_coordinates.py:151: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
../../.tox/py39-test-image-mpldev/lib/python3.9/site-packages/matplotlib/_api/deprecation.py:199: in wrapper
    emit_warning()
../../.tox/py39-test-image-mpldev/lib/python3.9/site-packages/matplotlib/_api/deprecation.py:193: in emit_warning
    warn_deprecated(
../../.tox/py39-test-image-mpldev/lib/python3.9/site-packages/matplotlib/_api/deprecation.py:96: in warn_deprecated
    warn_external(warning, category=MatplotlibDeprecationWarning)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

message = MatplotlibDeprecationWarning("The key_press_event function was deprecated in Matplotlib 3.6 and will be removed two minor releases later. Use callbacks.process('key_press_event', KeyEvent(...)) instead.")
category = <class 'matplotlib._api.deprecation.MatplotlibDeprecationWarning'>

    def warn_external(message, category=None):
        """
        `warnings.warn` wrapper that sets *stacklevel* to "outside Matplotlib".
    
        The original emitter of the warning can be obtained by patching this
        function back to `warnings.warn`, i.e. ``_api.warn_external =
        warnings.warn`` (or ``functools.partial(warnings.warn, stacklevel=2)``,
        etc.).
        """
        frame = sys._getframe()
        for stacklevel in itertools.count(1):  # lgtm[py/unused-loop-variable]
            if frame is None:
                # when called in embedded context may hit frame is None
                break
            if not re.match(r"\A(matplotlib|mpl_toolkits)(\Z|\.(?!tests\.))",
                            # Work around sphinx-gallery not setting __name__.
                            frame.f_globals.get("__name__", "")):
                break
            frame = frame.f_back
>       warnings.warn(message, category, stacklevel)
E       matplotlib._api.deprecation.MatplotlibDeprecationWarning: The key_press_event function was deprecated in Matplotlib 3.6 and will be removed two minor releases later. Use callbacks.process('key_press_event', KeyEvent(...)) instead.

../../.tox/py39-test-image-mpldev/lib/python3.9/site-packages/matplotlib/_api/__init__.py:363: MatplotlibDeprecationWarning
=========================== short test summary info ============================
FAILED ../../.tox/py39-test-image-mpldev/lib/python3.9/site-packages/astropy/visualization/wcsaxes/tests/test_display_world_coordinates.py::TestDisplayWorldCoordinate::test_overlay_coords
FAILED ../../.tox/py39-test-image-mpldev/lib/python3.9/site-packages/astropy/visualization/wcsaxes/tests/test_display_world_coordinates.py::TestDisplayWorldCoordinate::test_cube_coords
FAILED ../../.tox/py39-test-image-mpldev/lib/python3.9/site-packages/astropy/visualization/wcsaxes/tests/test_display_world_coordinates.py::TestDisplayWorldCoordinate::test_cube_coords_uncorr_slicing
================== 3 failed, 483 passed, 5 skipped in 59.29s ===================
```

--- a/astropy/visualization/wcsaxes/tests/test_display_world_coordinates.py
+++ b/astropy/visualization/wcsaxes/tests/test_display_world_coordinates.py
@@ -1,15 +1,12 @@
 # Licensed under a 3-clause BSD style license - see LICENSE.rst
 import matplotlib.pyplot as plt
-import pytest
 from matplotlib.backend_bases import KeyEvent
 
 import numpy as np
 
 import astropy.units as u
 from astropy.coordinates import FK5, SkyCoord
-from astropy.io import fits
 from astropy.time import Time
-from astropy.utils.data import get_pkg_data_filename
 from astropy.visualization.wcsaxes.core import WCSAxes
 from astropy.wcs import WCS
 from astropy.coordinates import galactocentric_frame_defaults
@@ -41,12 +38,12 @@ class TestDisplayWorldCoordinate(BaseImageTests):
 
         # Test pixel coordinates
         event1 = KeyEvent('test_pixel_coords', canvas, 'w')
-        fig.canvas.key_press_event(event1.key, guiEvent=event1)
+        fig.canvas.callbacks.process('key_press_event', event1)
         string_pixel = ax._display_world_coords(0.523412, 0.523412)
         assert string_pixel == "0.523412 0.523412 (pixel)"
 
         event3 = KeyEvent('test_pixel_coords', canvas, 'w')
-        fig.canvas.key_press_event(event3.key, guiEvent=event3)
+        fig.canvas.callbacks.process('key_press_event', event3)
         # Test that it still displays world coords when there are no overlay coords
         string_world2 = ax._display_world_coords(0.523412, 0.518311)
         assert string_world2 == '0\xb029\'45" -0\xb029\'20" (world)'
@@ -62,7 +59,7 @@ class TestDisplayWorldCoordinate(BaseImageTests):
         fig.savefig(tmpdir.join('test2.png').strpath)
 
         event4 = KeyEvent('test_pixel_coords', canvas, 'w')
-        fig.canvas.key_press_event(event4.key, guiEvent=event4)
+        fig.canvas.callbacks.process('key_press_event', event4)
         # Test that it displays the overlay world coordinates
         string_world3 = ax._display_world_coords(0.523412, 0.518311)
 
@@ -79,7 +76,7 @@ class TestDisplayWorldCoordinate(BaseImageTests):
         fig.savefig(tmpdir.join('test3.png').strpath)
 
         event5 = KeyEvent('test_pixel_coords', canvas, 'w')
-        fig.canvas.key_press_event(event4.key, guiEvent=event4)
+        fig.canvas.callbacks.process('key_press_event', event5)
         # Test that it displays the overlay world coordinates
         string_world4 = ax._display_world_coords(0.523412, 0.518311)
 
@@ -96,7 +93,7 @@ class TestDisplayWorldCoordinate(BaseImageTests):
         fig.savefig(tmpdir.join('test4.png').strpath)
 
         event6 = KeyEvent('test_pixel_coords', canvas, 'w')
-        fig.canvas.key_press_event(event5.key, guiEvent=event6)
+        fig.canvas.callbacks.process('key_press_event', event6)
         # Test that it displays the overlay world coordinates
         string_world5 = ax._display_world_coords(0.523412, 0.518311)
 
@@ -121,7 +118,7 @@ class TestDisplayWorldCoordinate(BaseImageTests):
 
         # Test pixel coordinates
         event1 = KeyEvent('test_pixel_coords', canvas, 'w')
-        fig.canvas.key_press_event(event1.key, guiEvent=event1)
+        fig.canvas.callbacks.process('key_press_event', event1)
         string_pixel = ax._display_world_coords(0.523412, 0.523412)
         assert string_pixel == "0.523412 0.523412 (pixel)"
 
@@ -148,7 +145,7 @@ class TestDisplayWorldCoordinate(BaseImageTests):
 
         # Test pixel coordinates
         event1 = KeyEvent('test_pixel_coords', canvas, 'w')
-        fig.canvas.key_press_event(event1.key, guiEvent=event1)
+        fig.canvas.callbacks.process('key_press_event', event1)
         string_pixel = ax._display_world_coords(0.523412, 0.523412)
         assert string_pixel == "0.523412 0.523412 (pixel)"
 

